python/lsst/pipe/base/connectionTypes.py

```diff
@@ -57,6 +57,14 @@ class BaseConnection:
     doc: str = ""
     multiple: bool = False
 
+    deprecated: str | None = dataclasses.field(default=None, kw_only=True)
+    """A description of why this connection is deprecated, including the
```

Comment:
It seems a bug that we are not documenting `doc` above. It seems weird that the docs for this are here and not up with the docstring like the others. What does sphinx perfer? I know python does nothing with strings defined below an attribute and just drops them on the floor, so interactively the docs are unavailable. Can you make these docs consistent one way or another?

Reply:
Ah, I hadn't even noticed that the others had been documented earlier.  I usually think of the class doc's "Parameters" section as being for `__init__` arguments, not instance attributes, but I suppose for dataclasses there's (usually) not much distinction.  I'll just move this docstring up with the rest.

Reply:
In a dataclass they are all init args unless you play with that though right?

Reply:
Like I said, I am fine if you want to move them all there, but it is a bit sad because it wont show up with `help`

Reply:
Right.  My tendency is to add class `Parameters` to a dataclass only if I actually define an `__init__` but to always document the attributes.  I don't think that's obviously better or worse than only documenting them in class `Parameters` as you've done here, though maybe we should always be documenting both?  (Of course, duplication is then its own problem.)

Reply:
I have actually moved the new one to `Parameters`; I was just explaining my usual practice elsewhere.  I'd rather resolve the general question about which (if either) of us should adopt the other's practice outside this ticket, and just keep things closer to the way they are already are for now.